### DISCUSSION TITLE
Targets be affected by event spells

### DISCUSSION THREAD

**Discussion by GabrielTadachi** [AscensionGameDev/Intersect-Engine#1528] [Ideas]
**Is your feature request related to a problem? Please describe.**
I was doing some tests regarding spells, and I noticed that when I create a spell that triggers an event, the event is called only to the person who casted.

**Describe the solution you'd like**
Create an option when selecting a event spell, that when it hits, being any kind of target type, it triggers the event on the intended target (as long as they are other players, since npcs can`t run events).

**Additional context**
Example: I created the spell Expelliarmus, that is linked to an event that should disarm the target wand, and do some damage. However, the event is triggered on the person that used the spell, disarming the player, while the target only gets the damage intended.

I also tried to create a projectile, which triggers an event spell, however it does nothing when hit the target.

Spell config:
![image](https://user-images.githubusercontent.com/25730747/130696085-062e64f0-99ad-4a92-a905-0649c276302e.png)
![image](https://user-images.githubusercontent.com/25730747/130696330-7c51488f-1dbf-4a54-b878-62988a550827.png)

Just to be clear: the player who casted triggers expelliarmus event, and the person who gets hit by the spell gets the damage and silence stat.

**Comment by Cheshire92** [comment]
This could work in theory, but events would never trigger for npcs and only players.

Would probably need a tick box or two to configure if it'll run for the caster and/or target player to not break current functionality though.

**Comment by GabrielTadachi** [comment]
Hi Cherise, thanks for your feedback!

I need to say that i agree with you, actually it was my mistake not to say that i asked to put this feature only to work with other players, since i know that npcs works in a different way, and cant run events!

But, if you think that this can be made, i`m happy to know! :)

**Comment by WeylonSantana** [comment]
Should it have the option to choose for whom the event should trigger, who cast it? the target? or both?
